core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -190,6 +216,28 @@ protected ClusterStateUpdateResponse newResponse(boolean acknowledged) {
                     }
 
                     @Override
+                    public void onAllNodesAcked(@Nullable Throwable t) {
+                        // the cluster state that includes the newly created index has been acknowledged,
+                        // now wait for the write consistency number of shards to be allocated before returning,
+                        // as that is when indexing operations can take place on the newly created index
+                        final long timeToTimeout = request.ackTimeout().nanos() - (System.nanoTime() - startTimeNS);
```

Comment:
The time span sampled here is from before we submit cluster state update until acking completed. There are two timeouts however that come into play when submitting cluster state updates, ackTimeout and masterNodeTimeout. I would make things simpler here and proceed in the same way that we do in other places: just do waitOnShards() with masterNodeTimeout.